### PR TITLE
Fixed #34355 -- Deprecated passing positional arguments to BaseConstraint.

### PR DESCRIPTION
**Pull request by xavfernandez**
Cf https://code.djangoproject.com/ticket/34355
&
https://github.com/django/django/pull/16560#pullrequestreview-1305496392

### CODE REVIEW

**Comment by felixxm** [comment]
@xavfernandez Thanks for this patch :+1: Unfortunately it's a documented API so we have to deprecate it first (see b8738aea14446b267a47087b52b38a98b440a6aa for a similar deprecation).

File: django/db/models/constraints.py [hunk 1]
```diff
@@ -9,6 +10,7 @@
 from django.db.models.query_utils import Q
 from django.db.models.sql.query import Query
 from django.db.utils import DEFAULT_DB_ALIAS
+from django.utils.deprecation import RemovedInDjango51Warning
```
Comment (felixxm) [inline django/db/models/constraints.py:13]: Things deprecated in Django 5.1 will be remove in Django 6.0.
```suggestion
from django.utils.deprecation import RemovedInDjango60Warning
```

File: docs/releases/5.0.txt [hunk 2]
```diff
@@ -267,6 +267,10 @@ Miscellaneous
 * The ``DjangoDivFormRenderer`` and ``Jinja2DivFormRenderer`` transitional form
   renderers are deprecated.
 
+* Passing positional arguments  ``name`` and ``violation_error_message`` to
+  :class:`~django.db.models.BaseConstraint` is deprecated in favor of keyword-only
+  arguments.
```
Comment (felixxm) [inline docs/releases/5.0.txt:272]: Please add note to the `docs/internals/deprecation.txt`, e.g.
```
* Support for passing positional arguments to ``BaseConstraint`` will be
  removed.
```

File: docs/releases/5.0.txt [hunk 3]
```diff
@@ -267,6 +267,10 @@ Miscellaneous
 * The ``DjangoDivFormRenderer`` and ``Jinja2DivFormRenderer`` transitional form
   renderers are deprecated.
 
+* Passing positional arguments  ``name`` and ``violation_error_message`` to
+  :class:`~django.db.models.BaseConstraint` is deprecated in favor of keyword-only
```
Comment (felixxm) [inline docs/releases/5.0.txt:271]: Wrap docs at 79 chars.

File: docs/ref/models/constraints.txt [hunk 4]
```diff
@@ -48,7 +48,7 @@ option.
 ``BaseConstraint``
 ==================
 
-.. class:: BaseConstraint(name, violation_error_message=None)
+.. class:: BaseConstraint(*, name, violation_error_message=None)
```
Comment (felixxm) [inline docs/ref/models/constraints.txt:51]: `deprecated` annotation is missing, e.g.
```
    .. deprecated:: 5.0

        Support for passing positional arguments is deprecated.
```

**Review by felixxm — COMMENTED** [review]
@xavfernandez Thanks :+1:

File: django/db/models/constraints.py (hunk 1 above)
Comment (xavfernandez) [reply django/db/models/constraints.py:13]: :+1: but I thought it would land in version 5.0 ? (and thus should be removed in 5.2 ?)

**Review by felixxm — COMMENTED** [review]
@xavfernandez Thanks :+1:

I added test for passing positional arguments.

**Comment by xavfernandez** [comment]
Thanks for the guidance and quick merge @felixxm ! :+1: 

#16560 has been rebased :innocent:

**Pull request merged**